internal/lookout/repository/jobs_test.go

```diff
@@ -8,12 +8,63 @@ import (
 
 	"github.com/doug-martin/goqu/v9"
 	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/util/duration"
 
 	"github.com/G-Research/armada/internal/common/util"
 	"github.com/G-Research/armada/pkg/api"
 	"github.com/G-Research/armada/pkg/api/lookout"
 )
 
+
+func TestGetJobs_GetPending(t *testing.T) {
+	withDatabase(t, func(db *goqu.Database) {
+		jobStore := NewSQLJobStore(db, userAnnotationPrefix)
+		jobRepo := NewSQLJobRepository(db, &util.DefaultClock{})
+
+		pendingTime := someTime.Add(time.Second)
+
+		_ = NewJobSimulator(t, jobStore).
+			CreateJobAtTime(queue, someTime).
+			PendingAtTime(cluster, k8sId1, pendingTime)
+
+		jobInfos, err := jobRepo.GetJobs(ctx, &lookout.GetJobsRequest{Take: 10})
+		assert.NoError(t, err)
+		assert.Equal(t, 1, len(jobInfos))
+
+		jobInfo := jobInfos[0]
+
+		timeInState := time.Now().Sub(pendingTime)
+	        expectedDuration := duration.ShortHumanDuration(timeInState)
+		assert.Equal(t, expectedDuration, jobInfo.JobStateDuration)
+	})
+}
+
+
+func TestGetJobs_GetRunning(t *testing.T) {
```

Comment:
Can you add tests for the missing States? Queued, Cancelled

Reply:
I added tests for each timestamp we are considering but will double check.